server/src/models/ledger.rs

```diff
@@ -51,9 +51,16 @@ impl Ledger {
 
                 let ledger_response = reqwest::get(&ledger_url)
                     .await
-                    .with_context(|| format!("failed to fetch ledger from URL: {ledger_url}"))?;
+                    .with_context(|| format!("failed to fetch ledger from URL: {ledger_url}"));
+
+                if let Err(error) = ledger_response {
+                    eprintln!("Failed to fetch ledger from URL: {error:?}");
+
+                    return Err(error.into());
+                }
 
                 let ledger_bytes = ledger_response
+                    .expect("Failed to fetch ledger response")
```

Comment:
`.expect` can panic, even though this case it won't, and is generally discouraged.

Reply:
What's the alternative? unwrap()?

Reply:
@robinbb In this case yes, but the code has changed since, so I'm resolving this